### ISSUE TITLE
[bitnami/redis] Fail to uninstall Redis chart Helm - prestop-redis.sh exited with 137

### ISSUE THREAD

**Issue by vincent-herlemont** [bitnami/charts#8031]
**Which chart**:
Redis 15.4.2

**Describe the bug**

Fail to uninstall Redis chart helm. 

**To Reproduce**

1. Download and uncompress: [issue_redis.zip](https://github.com/bitnami/charts/files/7485969/issue_redis.zip) - [Gist Here](https://gist.github.com/vincent-herlemont/54e4619ed1a4bad5ca7a844a00100d67)
2. Install it using [K3d](https://k3d.io)
    ```bash
    # Create the cluster
    $> k3d cluster create -c ./k3d-config-dev.yaml
    # Build dependency
    $> helm dependency build
    # Install the chart
    $> helm install redis-issue .
    # Wait for this state:
    $> kubectl get pods -o wide
    NAME                                   READY   STATUS    RESTARTS       AGE     IP          NODE                      NOMINATED NODE   READINESS GATES
    redis-issue-nfs-server-provisioner-0   1/1     Running   0              3m37s   10.42.1.4   k3d-redis-issue-agent-0   <none>           <none>
    redis-node-0                           2/2     Running   1 (112s ago)   3m37s   10.42.2.2   k3d-redis-issue-agent-2   <none>           <none>
    redis-node-1                           2/2     Running   0              88s     10.42.0.3   k3d-redis-issue-agent-1   <none>           <none>
    $> helm uninstall redis-issue
    ```
3. Redis nodes are stuck on `Terminating` status:
    ```bash
    $> kubectl get pods -o wide
    NAME           READY   STATUS        RESTARTS        AGE     IP          NODE                      NOMINATED NODE   READINESS GATES
    redis-node-0   2/2     Terminating   1 (2m27s ago)   4m12s   10.42.2.2   k3d-redis-issue-agent-2   <none>           <none>
    redis-node-1   2/2     Terminating   0               2m3s    10.42.0.3   k3d-redis-issue-agent-1   <none>           <none>
    ```
4. See error on describe pod:
    ```bash
    $> kubectl describe pod redis-node-0
      ...
      Normal   Pulled             4m57s                   kubelet            Successfully pulled image "docker.io/bitnami/redis-sentinel:6.2.6-debian-10-r0" in 3.40686648s
      Normal   Created            4m57s                   kubelet            Created container sentinel
      Normal   Started            4m57s                   kubelet            Started container sentinel
      Warning  Unhealthy          4m14s (x5 over 4m35s)   kubelet            Liveness probe failed:
      Normal   Killing            4m14s                   kubelet            Container redis failed liveness probe, will be restarted
      Warning  Unhealthy          3m49s (x10 over 4m35s)  kubelet            Readiness probe failed:
      Warning  FailedPreStopHook  3m44s                   kubelet            Exec lifecycle hook ([/bin/bash -c /opt/bitnami/scripts/start-scripts/prestop-redis.sh]) for Container "redis" in Pod "redis-node-0_default(d7957ef6-067a-4776-99ac-cdd6c87c3e2b)" failed - error: command '/bin/bash -c /opt/bitnami/scripts/start-scripts/prestop-redis.sh' exited with 137: , message: ""
    ```

    <details>

        ```bash
        Name:                      redis-node-0
        Namespace:                 default
        Priority:                  0
        Node:                      k3d-redis-issue-agent-2/172.18.0.4
        Start Time:                Fri, 05 Nov 2021 13:13:25 +0100
        Labels:                    app.kubernetes.io/component=node
                                   app.kubernetes.io/instance=redis-issue
                                   app.kubernetes.io/managed-by=Helm
                                   app.kubernetes.io/name=redis
                                   controller-revision-hash=redis-node-5c8fc9f9df
                                   helm.sh/chart=redis-15.4.2
                                   statefulset.kubernetes.io/pod-name=redis-node-0
        Annotations:               checksum/configmap: 8c152de3007d2b9312f3ee486199cb801d4638fb97c1fdc99bb14a7273eaba8c
                                   checksum/health: eee47413b4ab32629b46a45fef5befaca28f6920bd46b52567ce3f4e99e8b307
                                   checksum/scripts: 564a95d8955df0f2fa3bb96f9e7e375819026ae271a8f25f51f9acf203f0a4a2
                                   checksum/secret: 0a61d1419a94808cf3018e43dad79a28e0c6eef41a6ae9939c513955d8d97694
        Status:                    Terminating (lasts 49s)
        Termination Grace Period:  30s
        IP:                        10.42.2.2
        IPs:
          IP:           10.42.2.2
        Controlled By:  StatefulSet/redis-node
        Containers:
          redis:
            Container ID:  containerd://ded13f7177023d0d392a1a56122ee0501a5e955d64d149f302e853c890226ce6
            Image:         docker.io/bitnami/redis:6.2.6-debian-10-r0
            Image ID:      docker.io/bitnami/redis@sha256:5a463a98091f3dab2cce7e01adc894805f05e6c95a818cad6d6adeb66fbd56ae
            Port:          6379/TCP
            Host Port:     0/TCP
            Command:
              /bin/bash
            Args:
              -c
              /opt/bitnami/scripts/start-scripts/start-node.sh
            State:          Running
              Started:      Fri, 05 Nov 2021 13:14:51 +0100
            Last State:     Terminated
              Reason:       Error
              Exit Code:    1
              Started:      Fri, 05 Nov 2021 13:13:34 +0100
              Finished:     Fri, 05 Nov 2021 13:14:51 +0100
            Ready:          True
            Restart Count:  1
            Liveness:       exec [sh -c /health/ping_liveness_local.sh 5] delay=20s timeout=5s period=5s #success=1 #failure=5
            Readiness:      exec [sh -c /health/ping_readiness_local.sh 5] delay=20s timeout=1s period=5s #success=1 #failure=5
            Environment:
              BITNAMI_DEBUG:             false
              REDIS_MASTER_PORT_NUMBER:  6379
              ALLOW_EMPTY_PASSWORD:      no
              REDIS_PASSWORD:            <set to the key 'redis-password' in secret 'redis'>  Optional: false
              REDIS_MASTER_PASSWORD:     <set to the key 'redis-password' in secret 'redis'>  Optional: false
              REDIS_TLS_ENABLED:         no
              REDIS_PORT:                6379
              REDIS_DATA_DIR:            /data
            Mounts:
              /data from redis-data (rw)
              /health from health (rw)
              /opt/bitnami/redis/etc from redis-tmp-conf (rw)
              /opt/bitnami/redis/mounted-etc from config (rw)
              /opt/bitnami/scripts/start-scripts from start-scripts (rw)
              /tmp from tmp (rw)
              /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-4vpjc (ro)
          sentinel:
            Container ID:  containerd://15d78bf55e6066d5fe4e9d2da8fcb1d7dccc02e4aa9750e1519173df70788071
            Image:         docker.io/bitnami/redis-sentinel:6.2.6-debian-10-r0
            Image ID:      docker.io/bitnami/redis-sentinel@sha256:213b4cb2a94252d27eb555fa2f368297ce8069c09ac7eeb60fb8482cbcdd5752
            Port:          26379/TCP
            Host Port:     0/TCP
            Command:
              /bin/bash
            Args:
              -c
              /opt/bitnami/scripts/start-scripts/start-sentinel.sh
            State:          Running
              Started:      Fri, 05 Nov 2021 13:13:38 +0100
            Ready:          True
            Restart Count:  0
            Liveness:       exec [sh -c /health/ping_sentinel.sh 5] delay=20s timeout=5s period=5s #success=1 #failure=5
            Readiness:      exec [sh -c /health/ping_sentinel.sh 5] delay=20s timeout=1s period=5s #success=1 #failure=5
            Environment:
              BITNAMI_DEBUG:               false
              REDIS_PASSWORD:              <set to the key 'redis-password' in secret 'redis'>  Optional: false
              REDIS_SENTINEL_TLS_ENABLED:  no
              REDIS_SENTINEL_PORT:         26379
            Mounts:
              /data from redis-data (rw)
              /health from health (rw)
              /opt/bitnami/redis-sentinel/etc from sentinel-tmp-conf (rw)
              /opt/bitnami/redis-sentinel/mounted-etc from config (rw)
              /opt/bitnami/scripts/start-scripts from start-scripts (rw)
              /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-4vpjc (ro)
        Conditions:
          Type              Status
          Initialized       True 
          Ready             True 
          ContainersReady   True 
          PodScheduled      True 
        Volumes:
          redis-data:
            Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
            ClaimName:  redis-data-redis-node-0
            ReadOnly:   false
          start-scripts:
            Type:      ConfigMap (a volume populated by a ConfigMap)
            Name:      redis-scripts
            Optional:  false
          health:
            Type:      ConfigMap (a volume populated by a ConfigMap)
            Name:      redis-health
            Optional:  false
          config:
            Type:      ConfigMap (a volume populated by a ConfigMap)
            Name:      redis-configuration
            Optional:  false
          sentinel-tmp-conf:
            Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
            Medium:     
            SizeLimit:  <unset>
          redis-tmp-conf:
            Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
            Medium:     
            SizeLimit:  <unset>
          tmp:
            Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
            Medium:     
            SizeLimit:  <unset>
          kube-api-access-4vpjc:
            Type:                    Projected (a volume that contains injected data from multiple sources)
            TokenExpirationSeconds:  3607
            ConfigMapName:           kube-root-ca.crt
            ConfigMapOptional:       <nil>
            DownwardAPI:             true
        QoS Class:                   BestEffort
        Node-Selectors:              pool=scale
        Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 5s
                                     node.kubernetes.io/unreachable:NoExecute op=Exists for 5s
        Events:
          Type     Reason             Age                     From               Message
          ----     ------             ----                    ----               -------
          Warning  FailedScheduling   5m28s                   default-scheduler  0/3 nodes are available: 3 pod has unbound immediate PersistentVolumeClaims.
          Warning  FailedScheduling   5m26s                   default-scheduler  0/3 nodes are available: 3 pod has unbound immediate PersistentVolumeClaims.
          Normal   Scheduled          5m9s                    default-scheduler  Successfully assigned default/redis-node-0 to k3d-redis-issue-agent-2
          Normal   Pulling            5m7s                    kubelet            Pulling image "docker.io/bitnami/redis:6.2.6-debian-10-r0"
          Normal   Pulled             5m1s                    kubelet            Successfully pulled image "docker.io/bitnami/redis:6.2.6-debian-10-r0" in 5.672701893s
          Normal   Created            5m1s                    kubelet            Created container redis
          Normal   Started            5m1s                    kubelet            Started container redis
          Normal   Pulling            5m1s                    kubelet            Pulling image "docker.io/bitnami/redis-sentinel:6.2.6-debian-10-r0"
          Normal   Pulled             4m57s                   kubelet            Successfully pulled image "docker.io/bitnami/redis-sentinel:6.2.6-debian-10-r0" in 3.40686648s
          Normal   Created            4m57s                   kubelet            Created container sentinel
          Normal   Started            4m57s                   kubelet            Started container sentinel
          Warning  Unhealthy          4m14s (x5 over 4m35s)   kubelet            Liveness probe failed:
          Normal   Killing            4m14s                   kubelet            Container redis failed liveness probe, will be restarted
          Warning  Unhealthy          3m49s (x10 over 4m35s)  kubelet            Readiness probe failed:
          Warning  FailedPreStopHook  3m44s                   kubelet            Exec lifecycle hook ([/bin/bash -c /opt/bitnami/scripts/start-scripts/prestop-redis.sh]) for Container "redis" in Pod "redis-node-0_default(d7957ef6-067a-4776-99ac-cdd6c87c3e2b)" failed - error: command '/bin/bash -c /opt/bitnami/scripts/start-scripts/prestop-redis.sh' exited with 137: , message: ""
        ```

    </details>

\
**Expected behavior**

Remove pods without error.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
v3.7.0+geeac838
```

- Output of `kubectl version`:

```
Client Version: v1.21.5+k3s2
Server Version: v1.22.2+k3s2
```

**Comment by migruiz4** [comment]
Hi @vincent-herlemont,

Thank you very much for the detailed description and steps to reproduce.

I think the issue is caused by the `nfs-server-provisioner` being installed in the same chart as Redis. The provisioner is stopping before redis pods, causing the `prestop-redis.sh` command to fail when running the command `redis-cli`.

I did a quick test installing both chart separately and the pods were removed successfully.
